Maple2.Server.Game/Util/ItemStatsCalculator.cs

```diff
@@ -446,13 +446,17 @@ private int GetItemVariationValue(BasicAttribute? basicAttribute = null, Special
             } else if (TableMetadata.ItemVariationTable.Rates.TryGetValue(basicAttribute.Value, out ItemVariationTable.Range<float>[]? rates)) {
                 foreach (ItemVariationTable.Range<float> range in rates) {
                     if (rate >= range.Min && rate <= range.Max) {
-                        int convertedVariation = (int) range.Variation * 1000;
-                        int convertedRate = (int) rate * 1000;
+                        int convertedVariation = (int) (range.Variation * 1000);
+                        int convertedRate = (int) (rate * 1000);
                         return Random.Shared.Next(convertedRate, convertedRate + convertedVariation + 1);
                     }
                 }
             }
-        } else if (specialAttribute != null) {
+
+            return value;
```

Comment:
We're assuming here that the basicattribute is always a value, which is not true. it can be a rate too.